### DISCUSSION TITLE
Is there a list of methods I can use?

### DISCUSSION THREAD

**Discussion by Kistras** [FabricMC/fabric-api#1677] [FAQ] [answered]
When I needed to know how to summon an entity for my mod I've ended asking for help in Fabric's discord where people pointed out at `world.spawnEntity` function. Which made me wondering - is there any list of methods that I would be able to use? This way I (and probably any other people) wouldn't run in #mod-dev channels every time I needed to do something unspecified in tutorial and won't annoy actual developers.
Edit: I've tried searching for it and ended up with not minecraft Fabric, but something else python related (picture below). That's how I imagined list of methods. Do we have anything similar?
![image](https://user-images.githubusercontent.com/81854722/131224636-fadb2c39-964c-4886-83f2-6190eaa19ee3.png)

**Comment by warjort** [comment, marked as answer]
> Which made me wondering - is there any list of methods

Mojang don't publish javadocs for Minecraft :-)
They even obfuscate the code. The names you are using come from mod developers giving them names.
https://github.com/FabricMC/yarn/tree/1.17.1/mappings/net/minecraft

The python thing you found is a different project named fabric, not FabricMC.

**Reply by Kistras** [reply]
>https://github.com/FabricMC/yarn/tree/1.17.1/mappings/net/minecraft

Thank you for the mapping, that'll do. Not sure how to properly use it but I'll figure it out.
> Mojang don't publish javadocs for Minecraft :-)
> They even obfuscate the code. The names you are using come from mod developers giving them names.

Fabric isn't developed by Mojang either as far as I know. All those names came from mod developers and this is fine, but maybe someone have created a list where all of that is already mapped into a format similar to screenshot. That's the thing I was familiar with and that's what I was searching for.
> The python thing you found is a different project named fabric, not FabricMC.

I initially noted that this thing isn't related to minecraft in any way and needed this screenshot for reference.

Thanks.

**Reply by ghost** [reply]
Funny enough fabric does publish yarn mapped java docs for minecraft, heres link for latest 1.17.1 javadocs: https://maven.fabricmc.net/docs/yarn-1.17.1+build.46/index.html

**Reply by Kistras** [reply]
> Funny enough fabric does publish yarn mapped java docs for minecraft, heres link for latest 1.17.1 javadocs: https://maven.fabricmc.net/docs/yarn-1.17.1+build.46/index.html

Thank you!

**Comment by warjort** [comment]
You know you can run
````gradlew gensources````
to make the deobfuscated minecraft sources available in your ide? 

The source will have the names from the yarn files linked above.

**Reply by Kistras** [reply]
Now I know, thank you.
